### ISSUE TITLE
Pylint false reporting function not callable

### ISSUE THREAD

**Issue by rupakbajgain** [PyCQA/pylint#3845]
https://github.com/rupakbajgain/BLCalc/runs/1133293866 , it says blcalc\excel_load.py:140:39: E1102: generator.next is not callable (not-callable),

The generator is class,
```
class TempNameGenerator:
    """
    Used to create temp names
    """
    def __init__(self, start):
        """
        Start: name start from
        """
        self.count=0
        self.start=start

    def next(self):
        """
        Returns next string
        """
        self.count += 1
        if self.count == 1:
            return self.start
        return self.start + str(self.count)
```

**Comment by rupakbajgain** [comment]
I am using it as
a = TempNameGenerator('test')
print(a.next())
print(a.next())

**Comment by hippo91** [comment]
@rupakbajgain thanks for the report, however i'm not able to reproduce the bug. 
When linting the following lines:
```
class TempNameGenerator:
    """
    Used to create temp names
    """
    def __init__(self, start):
        """
        Start: name start from
        """
        self.count=0
        self.start=start

    def next(self):
        """
        Returns next string
        """
        self.count += 1
        if self.count == 1:
            return self.start
        return self.start + str(self.count)

a = TempNameGenerator('test')
print(a.next())
print(a.next())
```

i'have got the following messages:
```
bug_pylint_3845.py:1:0: C0114: Missing module docstring (missing-module-docstring)
bug_pylint_3845.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

Have you tried to upgrade your `pylint` version?

**Comment by rupakbajgain** [comment]
@hippo91 Sorry for that i did not search well,
 i tried that in same directory (inside package) the lint passed
it fails when i come out of directory and check lint

the line causing this error was from 
```from .helper import TempNameGenerator```
helper.py is the module containing this class

Looks the `.` in `.helper` is resonsible, though this shows in diffrent line,

**Comment by rupakbajgain** [comment]
& @hippo91 
i think for github actions use
```
      run: |
        pylint $(git ls-files '*.py')
```

as i was having problem too with that default as it don't search recursively.

**Comment by rupakbajgain** [comment]
& @hippo91 
suppose i am on linux
how can i do 
```
import os

if os.name == 'nt':
     import pywin32
```

**Comment by rupakbajgain** [comment]
Looks like it is related to #3651 ,
should i start new issue for os specific check as in previous command or it previously exists?

**Comment by hippo91** [comment]
@rupakbajgain i'm sorry but if i clearly understand the problem at the origin of this issue was not a bug.
That's why i will close this issue. If you are facing another problem please open another ticket. Thanks again.

**Comment by rupakbajgain** [comment]
@hippo91 ok thanks
